src/main/groovy/projet/gl51/store/MemoryProductStorage.groovy

```diff
@@ -2,28 +2,46 @@ package projet.gl51.store
 
 class MemoryProductStorage implements ProductStorage {
 
-    @Override
-    void save(Product p) {
+    ArrayList<Product> products
```

Comment:
Pourquoi pas une map ????
tu vas devoir parcourir la liste dans sa totalité pour récupérer un article.... complexité en o(n) au lieu de o(1) avec un hashmap !
http://javabypatel.blogspot.com/2015/10/time-complexity-of-hashmap-get-and-put-operation.html